# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../clucene/clucene-multimap-put.patch
# Copyright (C) 2013 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

--- clucene-core-2.3.3.4/src/core/CLucene/util/VoidMap.h
+++ clucene-core-2.3.3.4/src/core/CLucene/util/VoidMap.h
@@ -316,6 +316,7 @@ public:
		if ( _this::dk || _this::dv )
			_this::remove(k);

+		(*this)[k] = v;
	}
 };
